internal/provider/models/deployment.go

```diff
@@ -303,54 +295,112 @@ type HibernationSchedule struct {
 	WakeAtCron      types.String `tfsdk:"wake_at_cron"`
 }
 
+func HibernationStatusTypesObject(
+	ctx context.Context,
+	hibernationStatus *platform.DeploymentHibernationStatus,
+) (types.Object, diag.Diagnostics) {
+	if hibernationStatus == nil {
+		return types.ObjectNull(schemas.HibernationStatusAttributeTypes()), nil
+	}
+
+	obj := HibernationStatus{
+		IsHibernating: types.BoolValue(hibernationStatus.IsHibernating),
+		NextEventType: types.StringPointerValue((*string)(hibernationStatus.NextEventType)),
+		NextEventAt:   types.StringPointerValue(hibernationStatus.NextEventAt),
+		Reason:        types.StringPointerValue(hibernationStatus.Reason),
+	}
+	return types.ObjectValueFrom(ctx, schemas.HibernationStatusAttributeTypes(), obj)
+}
+
+func HibernationOverrideTypesObject(
+	ctx context.Context,
+	hibernationOverride *platform.DeploymentHibernationOverride,
+) (types.Object, diag.Diagnostics) {
+	if hibernationOverride == nil {
+		return types.ObjectNull(schemas.HibernationOverrideAttributeTypes()), nil
+	}
+	obj := HibernationSpecOverride{
+		IsHibernating: types.BoolPointerValue(hibernationOverride.IsHibernating),
+		IsActive:      types.BoolPointerValue(hibernationOverride.IsActive),
```

Comment:
Quick note: `IsActive` is a property set on the API response, it's not something that can be set on the request.

Reply:
Yep! got confused a bit by this sometimes but this is for creating the "Terraform response" similar to other "computed" values we have like `updatedAt`
In resource_deployment.go, we create the request where `isActive` is not there